docs/tutorials/building-your-first-bot.mdx

```diff
@@ -0,0 +1,346 @@
+    {
+        // Loads the bot prefab and spawns it into the scene
+        var bot = Instantiate(rgBotPrefab, botSpawnPoint.position, Quaternion.identity);
+        return bot;
+    }
+    
+    void Start()
+    {
+        // Prepares the RG SDK for spawning bots
+        RGBotServerListener.GetInstance()?.StartGame();
+        RGBotServerListener.GetInstance()?.SpawnBots();
+    }
+
+    private void OnDestroy()
+    {
+        // Stops all bots and cleans up the RG SDK
+        RGBotServerListener.GetInstance()?.StopGame();
+    }
+
+}
+```
+
+:::info
+
+Some bots may need additional information such as a character class, starting items, or other special spawn logic. 
+See the [full reference for seating and spawning bots](/integrating-with-unity/seating-and-spawning-bots) for 
+additional configuration options.
+
+:::
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to the `RGBotSpawner`. Position the `RGBotSpawner` within the scene wherever you want your bot to spawn 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**). Drag `RGBotSpawner`'s own Transform into the 
+**Bot Spawn Point** property of the RGBotSpawner's `CharacterBotSpawnManager` component. Finally, find the 
+**Player** prefab under `Assets > FirstBotDemo > Demo > Prefabs` and drag 
+it into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the 
+scene.
+
+![Screenshot of the RGOverlayCanvas.](./img/building-your-first-bot/spawner_6.png)
+
+## Define the game state
+
+We are onto the final parts of the integration! We now need to indicate what information the bot has access to in the game 
+(referred to here as the game's "state"), and which actions the bot can perform.
+
+First, let's define some state information. Regression Games marks states for bot consumption using the 
+[`[RGState]` attribute](/integrating-with-unity/defining-states#3---rgstate-fieldpropertymethod-attribute).
+This attribute can be applied to any public method that returns a value or any variable within a `MonoBehaviour`.
+Any values designated this way are automatically passed to your bots.
+
+First, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file, and add the following method to the class. This
+will allow our bot to know when it is currently in an attack animation (make sure to import any missing imports, with
+`using RegressionGames;`):
+
+```csharp
+[RGState]
+public bool IsAttacking()
+{
+    return !string.IsNullOrEmpty(_attackAnimation) && _animator.GetBool(_attackAnimation);
+}
+```
+
+Then, within the `Assets/FirstBotDemo/Demo/Scripts/EnemyController.cs` file, add the following attributes to the existing `GetCurrentHp()`
+and `GetTotalHp()` methods:
+
+```csharp
+// Near line 18
+[RGState("CurrentHealth")]
+private int _currentHp;
```

Comment:
Sounds like this needs to be public now

Reply:
@RG-nAmKcAz not sure if you saw my comment about this in the PR - would this work? Or does this need to be public? I wasn't sure with the new changes.

Reply:
I see now, I missed the public part in the docs.

Reply:
Let me know what you think of this... I think having them change it to public is good so they learn that it has to be, but also now the naming scheme is inconsistent with the `_`. Maybe that's ideal/best practice? i.e. maybe that means "this is public but only because RG wants to see it - treat it as private!" Not sure... looking for opinions.

Reply:
I don't like the idea of asking people to either violate naming conventions or rename the variable and update all occurrences of it in the project. They can if they want to, but I think this is where we suggest introducing a new readonly variable to expose that data